openhands/integrations/github/github_service.py

```diff
@@ -537,6 +540,86 @@ async def create_pr(
         # Return the HTML URL of the created PR
         return response['html_url']
 
+    async def get_microagents(self, repository: str) -> list[MicroagentResponse]:
+        """Fetch microagents from GitHub repository using GitHub Contents API.
+
+        Args:
+            repository: Repository name in format 'owner/repo'
+
+        Returns:
+            List of microagents found in the repository (without content for performance)
+        """
+        microagents_path = self._determine_microagents_path(repository)
+        base_url = f'{self.BASE_URL}/repos/{repository}/contents'
+        microagents = []
+
+        # Step 1: Check for .cursorrules file
+        try:
+            cursorrules_response, _ = await self._make_request(
+                f'{base_url}/.cursorrules'
+            )
+            if cursorrules_response:
+                microagents.append(
+                    self._create_microagent_response('.cursorrules', '.cursorrules')
+                )
+        except ResourceNotFoundError:
+            logger.debug(f'No .cursorrules file found in {repository}')
+        except Exception as e:
+            logger.warning(f'Error checking .cursorrules file in {repository}: {e}')
```

Comment:
let's see if we can modularize this more

can we make a method `get_microagents ` part of `BaseGitService` class, and pass the api routes as args (something like that)

Similar logic seems to appear across all the service classes; same goes for fetching the microagent directory just beneath this piece of code

Reply:
actually i do see small variations between the service classes when processing the response from the api - see if its possible to dedup the logic maybe but if not its ok